hulks/check_mutable_defaults.py

```diff
@@ -0,0 +1,66 @@
+import ast
+import sys
+
+from hulks.base import BaseHook
+
+
+class CheckMutableDefaults(BaseHook):
+    _immutable_builtins = (
+        'bool',
+        'float',
+        'frozenset',
+        'int',
+        'object',
+        'str',
+        'tuple',
+    )
+    _ast_mutable_types = (
+        ast.List,
+        ast.Set,
+        ast.Dict,
+    )
+
+    def _collect_functions_with_defaults(self, tree):
+        nodes = []
+        for node in ast.walk(tree):
+            if not isinstance(node, ast.FunctionDef):
+                continue
+
+            if node.args and node.args.defaults:
+                nodes.append(node)
+
+        return nodes
+
+    def _check_mutable_defaults(self, filename, node):
+        retval = True
+        for default_arg_value in node.args.defaults:
+            conditions = (
```

Comment:
kinda of a :lipstick: on a :pig:, but the code might be easier to read if a method `_is_mutable` is added:

```python
for default_arg_value in node.args.defaults:
    if self._is_mutable(default_arg_value):
        msg = ...
```